# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1263449487 18000
#      Thu Jan 14 01:11:27 2010 -0500
# Node ID 8b4e3388a25412f8e5d351fd70c293d71985f34a
# Parent  b51848e95e4bf98f0526b331e12eefc6133baf35
oct-rl-edit.c (octave_rl_set_terminal_name): save term string before setting rl_terminal_name

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2010-01-14  John W. Eaton  <jwe@octave.org>
+
+	* oct-rl-edit.c (octave_rl_set_name): Save arg before setting
+	rl_terminal_name.
+
 2010-01-13  John W. Eaton  <jwe@octave.org>
 
 	* idx-vector.cc (idx_vector::idx_scalar_rep::idx_scalar_rep (T x)):
 	Initialize dummy variable.
 
 2010-01-13  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Array.cc (Array<T>::compute_index): Generalize to the case of
diff --git a/liboctave/oct-rl-edit.c b/liboctave/oct-rl-edit.c
--- a/liboctave/oct-rl-edit.c
+++ b/liboctave/oct-rl-edit.c
@@ -341,17 +341,19 @@ void
 octave_rl_add_defun (const char *name, rl_fcn_ptr f, char key)
 {
   rl_add_defun (name, f, key);
 }
 
 void
 octave_rl_set_terminal_name (const char *term)
 {
-  rl_terminal_name = term;
+  OCTAVE_RL_SAVE_STRING (saved_term, term);
+
+  rl_terminal_name = saved_term;
 }
 
 void
 octave_rl_initialize (void)
 {
   rl_initialize ();
 }
 
